python/ray/serve/controller.py

```diff
@@ -364,8 +364,13 @@ def deploy(
 
         autoscaling_config = deployment_config.autoscaling_config
         if autoscaling_config is not None:
-            # TODO: is this the desired behaviour? Should this be a setting?
-            deployment_config.num_replicas = autoscaling_config.min_replicas
+            previous_deployment = self.deployment_state_manager.get_deployment(name)
+            if previous_deployment is None:
+                deployment_config.num_replicas = autoscaling_config.min_replicas
+            else:
+                deployment_config.num_replicas = (
```

Comment:
Can we change this to

```python
deployment_config.num_replicas = min(
    max(
        previous_deployment.deployment_config.num_replicas,
        autoscaling_config.min_replicas
    ), 
    autoscaling_config.max_replicas
)
```

This would maintain the invariant that `num_replicas` is between `min_replicas` and `max_replicas`.

Reply:
I'm pretty sure autoscaling loop will ensure this invariant, so we don't need to do anything to interfere with it. 

https://github.com/ray-project/ray/blob/98da294f0b8fbf0ab9d94ebd552c4d78b4d3aaf8/python/ray/serve/_private/autoscaling_policy.py#L47-L49

Reply:
Ah good point, probably good to not do this in multiple places.  Agree we should add a test for this scenario